### PR TITLE
dggaytan/DCGAN

### PR DESCRIPTION
**Pull request by dggaytan**
- Enabled XPU training in main.py
- Added XPU flag in README.md file

### CODE REVIEW

File: dcgan/README.md [hunk 1]
```diff
@@ -20,31 +20,38 @@ python download.py -c bedroom
...
-  --ndf NDF             number of filters in the discriminator
+  --ngf NGF
+  --ndf NDF
   --niter NITER         number of epochs to train for
   --lr LR               learning rate, default=0.0002
   --beta1 BETA1         beta1 for adam. default=0.5
   --cuda                enables cuda
-  --mps                 enables macOS GPU
+  --xpu                 enables XPU training
+  --dry-run             check a single training cycle works
```
Comment (jafraustro) [inline dcgan/README.md:48]: do you add the `--dry-run`  option?

File: dcgan/main.py [hunk 2]
```diff
@@ -280,4 +293,4 @@ def forward(self, input):
             break
     # do checkpointing
     torch.save(netG.state_dict(), '%s/netG_epoch_%d.pth' % (opt.outf, epoch))
-    torch.save(netD.state_dict(), '%s/netD_epoch_%d.pth' % (opt.outf, epoch))
+    torch.save(netD.state_dict(), '%s/netD_epoch_%d.pth' % (opt.outf, epoch))
```
Comment (jafraustro) [inline dcgan/main.py:296]: add empty line

File: dcgan/README.md [hunk 3]
```diff
@@ -20,31 +20,38 @@ python download.py -c bedroom
...
-  --batchSize BATCHSIZE input batch size
-  --imageSize IMAGESIZE the height / width of the input image to network
+  --batchSize BATCHSIZE
+                        input batch size
+  --imageSize IMAGESIZE
+                        the height / width of the input image to network
   --nz NZ               size of the latent z vector
-  --ngf NGF             number of filters in the generator
-  --ndf NDF             number of filters in the discriminator
+  --ngf NGF
```
Comment (jafraustro) [inline dcgan/README.md:41]: missing description

File: dcgan/README.md [hunk 4]
```diff
@@ -20,31 +20,38 @@ python download.py -c bedroom
...
   --cuda                enables cuda
-  --mps                 enables macOS GPU
+  --xpu                 enables XPU training
+  --dry-run             check a single training cycle works
   --ngpu NGPU           number of GPUs to use
   --netG NETG           path to netG (to continue training)
   --netD NETD           path to netD (to continue training)
   --outf OUTF           folder to output images and model checkpoints
-  --manualSeed SEED     manual seed
+  --manualSeed MANUALSEED
```
Comment (jafraustro) [inline dcgan/README.md:53]: there is an issue with the format.

**Review by jafraustro — CHANGES_REQUESTED** [review]

File: dcgan/README.md (hunk 1 above)
Comment (dggaytan) [reply dcgan/README.md:48]: I did not add it, it was there already

File: dcgan/README.md (hunk 4 above)
Comment (dggaytan) [reply dcgan/README.md:53]: this is a trunc limit, due to the 80 chars per line requirement,
`--manualSeed MANUALSEED
                        manual seed`
do you approve to change "MANUALSEED" to "SEED" to fit the overall format?

Comment (jafraustro) [reply dcgan/README.md]: We can keep as it appears in the main branch

File: dcgan/README.md [hunk 5]
```diff
@@ -20,31 +20,37 @@ python download.py -c bedroom
...
+               [--mps]
+
+options:
   -h, --help            show this help message and exit
   --dataset DATASET     cifar10 | lsun | mnist |imagenet | folder | lfw | fake
   --dataroot DATAROOT   path to dataset
   --workers WORKERS     number of data loading workers
-  --batchSize BATCHSIZE input batch size
-  --imageSize IMAGESIZE the height / width of the input image to network
+  --batchSize BATCHSIZE
```
Comment (framoncg) [inline dcgan/README.md:36]: is this pure aesthetic? as in a linter change?

File: dcgan/README.md [hunk 6]
```diff
@@ -20,31 +20,37 @@ python download.py -c bedroom
...
+                        input batch size
+  --imageSize IMAGESIZE
+                        the height / width of the input image to network
   --nz NZ               size of the latent z vector
-  --ngf NGF             number of filters in the generator
-  --ndf NDF             number of filters in the discriminator
+  --ngf NGF             number of generator filters
```
Comment (framoncg) [inline dcgan/README.md:41]: same here, the previous implementation seems clear

File: dcgan/README.md (hunk 5 above)
Comment (dggaytan) [reply dcgan/README.md:36]: yes, it was changed due to the maximum char per line of 80

File: dcgan/README.md (hunk 6 above)
Comment (dggaytan) [reply dcgan/README.md:41]: I don't get the requested change, can you elaborate on what is needed to change here?

File: dcgan/README.md (hunk 5 above)
Comment (framoncg) [reply dcgan/README.md:36]: I don't know this will pass as a change, I'm okay with it but sometimes this changes clash with the linters on the repo we are making the PR

File: dcgan/README.md (hunk 6 above)
Comment (framoncg) [reply dcgan/README.md:41]: I get that this change was made because of the limit of chars per line?

**Review by jafraustro — APPROVED** [review]
LFTM
